source/code/include/playfab/PlayFabError.h.ejs

```diff
@@ -42,4 +42,29 @@ namespace PlayFab
 
     typedef std::function<void(const PlayFabError& error, void* customData)> ErrorCallback;
     typedef std::function<void(std::exception exception)> ExceptionCallback;
+
+    enum class PlayFabExceptionCode
+    {
+        AuthContextRequired,
+        DeveloperKeyNotSet,
+        EntityTokenNotSet,
+        NotLoggedIn,
+        PluginAmbiguity,
+        PluginNotFound,
+        ThreadMisuse,
+        TitleNotSet,
+    };
+
+    class PlayFabException : public std::runtime_error
+    {
+    public:
+        PlayFabException() = delete;
+        PlayFabException(const PlayFabException& source) = default;
+        PlayFabException(PlayFabException&&) = default;
+        PlayFabException& operator=(const PlayFabException& source) = default;
+        PlayFabException& operator=(PlayFabException&& other) = default;
+
```

Comment:
for my own knowledge - why do we need these i.e - they're not marked as deleted ?

Reply:
The way we're throwing these, the copy constructor seems to be required. It's a compile error otherwise.
After that, nothing I'm doing is altering the way that runtime_error works, so I just want to let it keep working, with the one additional bit of info we're providing (PlayFabExceptionCode).

---
In reply to: [343317602](https://github.com/PlayFab/XPlatCoreTemplate/pull/39#discussion_r343317602) [](ancestors = 343317602)

Reply:
You should provide explicit deletion of any of the copy control operators that you don't need

---
In reply to: [343340226](https://github.com/PlayFab/XPlatCoreTemplate/pull/39#discussion_r343340226) [](ancestors = 343340226,343317602)